File: tests/runtimes/temporal/temporal.py
# Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
# SPDX-License-Identifier: MPL-2.0

from typegraph import t, typegraph, Policy, Graph
from typegraph.providers.temporal import TemporalRuntime


@typegraph()
def temporal(g: Graph):
    public = Policy.public()
    temporal = TemporalRuntime(
        "<name>", "<host_secret>", namespace_secret="<ns_secret>"
    )
    arg = t.struct({"some_field": t.string()})

    g.expose(
        public,
        start=temporal.start_workflow("<workflow_type>", arg),
        query=temporal.query_workflow("<query_type>", arg, t.string()),
        signal=temporal.signal_workflow("<signal_name>", arg).reduce(
            {"workflow_id": "1234"}
        ),
        describe=temporal.describe_workflow(),
    )

File: tests/runtimes/temporal/temporal.ts
// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
// SPDX-License-Identifier: MPL-2.0

import { Policy, t, typegraph } from "@typegraph/sdk";
import { TemporalRuntime } from "@typegraph/sdk/providers/temporal";

export const tg = await typegraph("temporal", (g: any) => {
  const pub = Policy.public();
  const temporal = new TemporalRuntime({
    name: "test",
    hostSecret: "HOST",
    namespaceSecret: "NAMESPACE",
  });
  g.expose({
    startKv: temporal
      .startWorkflow("keyValueStore", t.struct({}))
      .withPolicy(pub),

    query: temporal
      .queryWorkflow("getValue", t.string(), t.string().optional())
      .withPolicy(pub),

    signal: temporal
      .signalWorkflow(
        "setValue",
        t.struct({ key: t.string(), value: t.string() }),
      )
      .withPolicy(pub),

    describe: temporal.describeWorkflow().withPolicy(pub),
  });
});
